include/deal.II/hp/fe_collection.h

```diff
@@ -495,6 +501,27 @@ namespace hp
   }
 
 
+
+  template <int dim, int spacedim>
+  inline
+  bool
+  FECollection<dim,spacedim>::operator== (const FECollection<dim,spacedim> &fe_collection) const
+  {
+    bool is_same = (size() == fe_collection.size());
+    if (is_same)
+      {
+        for (unsigned int i=0; i<size(); ++i)
+          if (!(*finite_elements[i] == fe_collection[i]))
```

Comment:
Hm, this currently only calls a non-`virtual` `operator==` in the `FiniteElement` case class. I wonder if that is going to call two elements the same that are really not. For example, the `FE_Nothing(1,true)` and `FE_Nothing(1,false)` are almost certainly going to be called the same, though they are not :-(

Reply:
>  the FE_Nothing(1,true) and FE_Nothing(1,false) are almost certainly going to be called the same, though they are not :-(

yeah... If we merge this PR, we should at least create an issue to investigate this issue later...